packages/react-reconciler/src/ReactFiber.new.js

```diff
@@ -511,7 +504,7 @@ export function createFiberFromTypeAndProps(
       case REACT_STRICT_MODE_TYPE:
         fiberTag = Mode;
         mode |= StrictLegacyMode;
-        if (enableStrictEffects && (mode & ConcurrentMode) !== NoMode) {
+        if ((mode & ConcurrentMode) !== NoMode) {
```

Comment:
Since `enableStrictEffects` used to be set to `__DEV__`, does this mean we flip this flag in prod more often (here and in other places where there's no `__DEV__` check). Does this matter? It seems good for consistency to always set `StrictLegacyMode` together with `StrictEffectsMode` but I want to make sure there's no accidental assumptions about which flags are DEV-only.

Reply:
I don't think it matters. StrictMode itself has a check to see if `__DEV__` is true or not. So even if this flag gets flipped more often than before, it will not cause StrictMode to be triggered in prod.
I'm trying to reduce complexity around StrictMode, it seems unnecessary to have several flags controlling it, we want to behaviour to be consistent.